### ISSUE TITLE
Publish Dataset: Unable to publish datasets that use handles due to missing globalidcreatetime from migration.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#3081]
See #3079 for reference.

Unpublished datasets that use handles currently cannot be published since the globalidcreatetime is blank. This field is used as a flag by the 4.0 code to indicate the identifier was actually registered. Since we are not yet supporting creating handles, we cannot attempt to publish -it tries to publish as DOI and fails.

Additionally, in v3.x these identifiers were created through handle.net upon creating the initial draft, so more than likely it is OK to set dataset.globalidcreatetime=dvobject.createdate. This would be a migration issue and I will add it to the list.

One last consideration is that the original handle registration process was less robust in error checking so in some cases the handles may not be properly registered. That needs to be verified and corrected as part of updating globalidcreatetime. 

So to summarize:
1. A migration issue for unpublished datasets created using handles prevents them from being published.
2. A workaround is to set dataset.globalidcreatetime=dvobject.createdate, this works since the handle was already registered and there is code in place that correctly processes existing handle identifiers.
3. Before we do this for all unpublished handles, we need to identify and correct handles that were not properly registered but not caught by the earlier code.

**Comment by djbrooke** [comment]
Closed #4046 as a duplicate. It had some stats:

For 2251 out of 11,000+ in the 1902.1 namespace there was no publicationdate and no globalidcreatetime.
This will prevent them from being published.

**Comment by djbrooke** [comment]
Closed #3509 as well

**Comment by sekmiller** [comment]
Here are the update queries to allow migrated handles datasets to be published.  (Note that there are two different queries - one to be applied before File DOIs are merged and the other for after.):


/*before file DOI is merged*/
update dataset set dataset.globalidcreatetime=dvobject.createdate where dataset.id = dvbject.id and 
 authority = '1902.1' and identifier is not null and globalidcreatetime is null;

/*AFTER file DOI is merged*/
 update dvobject set globalidcreatetime=createdate where 
 authority = '1902.1' and identifier is not null and globalidcreatetime is null;

**Comment by sekmiller** [comment]
Fixed the joins for the case where the update is run prior to the file DOI merge:




UPDATE dataset  SET  globalidcreatetime=dvo.createdate
  FROM dataset AS ds
  INNER JOIN dvobject AS dvo
  ON ds.id = dvo.id
  WHERE ds.authority = '1902.1' and ds.identifier is not null and ds.globalidcreatetime is null;

**Comment by kcondon** [comment]
I've run the updated statement in a transaction on thedata_alt and it appears to incorrectly update all datasets.

**Comment by sekmiller** [comment]
This update query only updates those rows that need to be updated:

UPDATE dataset
SET    globalidcreatetime = dvobject.createdate
FROM   dvobject
WHERE  dataset.id = dvobject.id
       AND dataset.authority = '1902.1' and dataset.identifier is not null and dataset.globalidcreatetime is null;
